### ISSUE TITLE
Debian 12 (bookworm) support

### ISSUE THREAD

**Issue by jlaundry** [Azure/azure-functions-core-tools#3431]
Hi folks,

Now that bookworm has been out for a month, people are starting to update their toolchains, and have found that the `azure-functions-core-tools` deb package is missing from `https://packages.microsoft.com/debian/12/prod/pool/main/a/` (https://github.com/jlaundry/devcontainer-features/issues/12).

Is there a timeframe for when Debian 12 will be supported, with .deb release?

**Comment by shahafal** [comment]
+1

anyone tried to run the Debian 11 azure-functions-core-tools on Debian 12 and can share if this is a legitimate workaround until the official Debian 12 package is released?

**Comment by shibayan** [comment]
+1

This is very important because if Azure Functions Core Tools support for bookworm is delayed, it will not be possible to develop using .NET 8, only bookworm is released as an official image, and bullseye is no longer supported.

**Comment by pl4nty** [comment]
+1. looks like `mcr.microsoft.com/azure-functions/powershell:4-powershell7.4-core-tools` is broken from this, since it depends on `mcr.microsoft.com/dotnet/sdk:8.0` which only has bookworm images

**Comment by IdeaHunter** [comment]
Please raise level of ticket, dotnet 6.0 expires this year and there is still no linux support for azure functions together with net 8.0 on linux

**Comment by iddqdiddqd** [comment]
+1

It's Feb 2024 and `azure-functions-core-tools-4` is still not supported on bookworm

**Comment by jlaundry** [comment]
😮

Surprise! What's this? https://packages.microsoft.com/debian/12/prod/pool/main/a/azure-functions-core-tools/

Looks like 12 (bookworm) support was silently added in 4.0.5571

(and the change broke Debian 11, as 11 doesn't support zst compression: https://github.com/Azure/azure-functions-core-tools/issues/3626)

**Comment by mverbaas** [comment]
It seems like the package isn't properly signed

```
Hit:1 http://deb.debian.org/debian bookworm InRelease
Hit:2 http://deb.debian.org/debian bookworm-updates InRelease                                         
Hit:3 http://deb.debian.org/debian-security bookworm-security InRelease                               
Get:4 https://packages.microsoft.com/debian/12/prod bookworm InRelease [3,618 B]                      
Hit:5 https://dl.yarnpkg.com/debian stable InRelease       
Err:4 https://packages.microsoft.com/debian/12/prod bookworm InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY EB3E94ADBE1229CF
Reading package lists... Done
W: GPG error: https://packages.microsoft.com/debian/12/prod bookworm InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY EB3E94ADBE1229CF
E: The repository 'https://packages.microsoft.com/debian/12/prod bookworm InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package azure-functions-core-tools-4
```

**Comment by jlaundry** [comment]
@mverbaas it looks like you need to add the Microsoft key first:

```bash
curl -fsSL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/microsoft.gpg > /dev/null
```

Or, if you prefer to limit the key to a specific apt source,

```bash
curl -fsSL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | sudo tee /usr/share/keyrings/packages.microsoft.gpg > /dev/null

echo "deb [signed-by=/usr/share/keyrings/packages.microsoft.gpg] https://packages.microsoft.com/debian/12/prod bookworm main" | sudo tee /etc/apt/sources.list.d/microsoft-prod.list >/dev/null
```
